### PR TITLE
fix(python_samples): README link fix, enforce samples=True 

### PR DESCRIPTION
**Pull request by runargs**
Updated sources link to direct to specified product's repo, enforced `kwargs["metadata"]["samples"] = True` every time `py_samples()` is run, as it's a given if the function is called and shouldn't have to be specified from the user side.
Oh also the wording was slightly redundant.
TLDR, Just some wording fixes and a one line edit

### CODE REVIEW

**Review by kolea2 — APPROVED** [review]

File: synthtool/gcp/common.py [hunk 1]
```diff
@@ -90,6 +90,9 @@ def py_samples(self, **kwargs) -> List[Path]:
             ]
         )
 
+        # ensure samples will generate
+        kwargs["metadata"]["samples"] = True
```
Comment (kolea2) [inline synthtool/gcp/common.py:94]: what happens if this is not set?

File: synthtool/gcp/templates/python_samples/README.md [hunk 2]
```diff
@@ -37,15 +37,14 @@ To run this sample:
 {% if metadata['repo']|length %}{% if metadata['repo']['client_library'] %}
 These samples use the [Google Cloud Client Library for Python][client_library_python].{% endif %}
 You can read the documentation for more details on API usage and use GitHub
-to [browse the source][source] and [report issues][issues].{% endif %}
+to <a href="https://github.com/{{ metadata['repo']['repo'] }}">browse the source</a> and [report issues][issues].{% endif %}
```
Comment (kolea2) [inline synthtool/gcp/templates/python_samples/README.md:40]: can we add a quick test that the url is correct?

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: synthtool/gcp/common.py (hunk 1 above)
Comment (runargs) [reply synthtool/gcp/common.py:94]: if it isn't set, _generic_library() will add README.md to the `excludes` https://github.com/googleapis/synthtool/blob/ca275fc17e8e92c81c04f19de1d22e016be3f0e2/synthtool/gcp/common.py#L57

File: synthtool/gcp/templates/python_samples/README.md [hunk 3]
```diff
@@ -37,15 +37,14 @@ To run this sample:
...
 
 ### Contributing
-For [contributing guidelines][contrib_guide], the [Python style guide][py_style], and more information on prerequisite steps to contribute, view the source code at <a href="https://github.com/{{ metadata['repo']['repo'] }}">{{ metadata['repo']['repo'] }}</a>.
+View the [contributing guidelines][contrib_guide], the [Python style guide][py_style]. More information on prerequisite steps to contribute may be available at the source code.
```
Comment (kolea2) [inline synthtool/gcp/templates/python_samples/README.md:43]: I would just make this read, "View the contributing guidelines and the Python style guide for more information."

**Review by kolea2 — CHANGES_REQUESTED** [review]

**Review by kolea2 — APPROVED** [review]

**Pull request merged**